src/porepy/models/constitutive_laws.py

```diff
@@ -742,7 +742,93 @@ def normal_permeability(self, interfaces: list[pp.MortarGrid]) -> pp.ad.Operator
         return Scalar(self.solid.normal_permeability())
 
 
-class CubicLawPermeability(ConstantPermeability):
+class DimensionDependentPermeability(ConstantPermeability):
+    """Permeability depending on subdomain dimension.
+
+    The use of sub-methods allows convenient code reuse if the permeability
+    for one of the subdomain sets is changed.
+
+    """
+
+    nd: int
+    """Ambient dimension of the problem. Normally set by a mixin instance of
+    :class:`porepy.models.geometry.ModelGeometry`.
+
+    """
+
+    def permeability(self, subdomains: list[pp.Grid]) -> pp.ad.Operator:
+        """Permeability [m^2].
+
+        This function combines the permeability of the matrix, fractures and
+        intersections.
+
+        Parameters:
+            subdomains: List of subdomains.
+
+        Returns:
+            Cell-wise permeability values.
+
+        """
+        projection = pp.ad.SubdomainProjections(subdomains, dim=1)
+        matrix = [sd for sd in subdomains if sd.dim == self.nd]
+        fractures: list[pp.Grid] = [sd for sd in subdomains if sd.dim == self.nd - 1]
+        intersections: list[pp.Grid] = [sd for sd in subdomains if sd.dim < self.nd - 1]
+
+        permeability = (
+            projection.cell_prolongation(matrix) @ self.matrix_permeability(matrix)
+            + projection.cell_prolongation(fractures)
+            @ self.fracture_permeability(fractures)
+            + projection.cell_prolongation(intersections)
+            @ self.intersection_permeability(intersections)
+        )
+        return permeability
+
+    def matrix_permeability(self, subdomains: list[pp.Grid]) -> pp.ad.Operator:
+        """Permeability of the matrix.
+
+        Parameters:
+            subdomains: List of subdomains.
+
+        Returns:
+            Cell-wise permeability operator.
+
+        """
+        return super().permeability(subdomains)
+
+    def fracture_permeability(self, subdomains: list[pp.Grid]) -> pp.ad.Operator:
+        """Permeability of fractures.
```

Comment:
Does this method encompass both in-plane and out-plane permeabilities? If so, please clarify it.

Reply:
No, the out-of-plane permeability is handled in `normal_permeability`.